### DISCUSSION TITLE
Faster area of intersection computation?

### DISCUSSION THREAD

**Discussion by mattjohnsonpint** [NetTopologySuite/NetTopologySuite#489] [Q&A]
Hi.  In my code, I need to determine the area of a geometry that intersects an envelope.  I do this with:

```csharp
double area = feature.Geometry.Intersection(envelope).Area;
```

It works fine, but I have several hundred thousand iterations to perform, and this is where the CPU is spending the vast majority of its time.  I've decreased total run time by using all cores in parallel, via `Parallel.ForEach` in the function that calls this, but it is still super slow at scale.

Elsewhere in my code, I had a call to `geometry.Intersects(envelope)` that was also expensive, until I started using `PreparedGeometry.Intersects` instead.  That made a huge perf difference.  Unfortunately, there doesn't seem to be anything on `PreparedGeometry` that would help with area of intersection.

I don't actually need the geometry created by the intersection - I only need to know the area that intersects.  Is there any faster API I should be using for this?  Or perhaps there's some technique to apply ahead of time that would make this operation go faster in general?

I tried simplifying the geometries first, but it didn't seem to help.  The simplification was also expensive, and the computed areas were too different to be useful to me.

Any ideas would be greatly appreciated.  Thanks. 

(I'm using NTS 1.15.1 on .NET Core 2.1)

**Comment by FObermaier** [comment]
* You should use `PreparedGeometry.Intersects(envelope)` to filter out geometries that don't intersect and only compute the intersection for those that do.
* If `PreparedGeometry.Contains(envelope)` is `true`, the area is that of `envelope`

**Comment by mattjohnsonpint** [comment]
Yeah, I already made both of those optimizations.  Was hoping there was something more I could do.

Is there some way to ask for *approximate* area? Or some way to compute area of intersection without building the intersecting geometry?

**Comment by airbreather** [comment]
`PreparedGeometry.intersection()` has been on a list of "future enhancements" since at least as far back as [this PowerPoint from 2007](http://jump-pilot.sourceforge.net/pdfs/jts_secrets_foss4g2007.pdf)... I think we'd need something like that in order to do what you're looking for.

**Comment by gktval** [comment]
This is a bit of an older discussion; but in my case, I was able to perform a union on the first set of geometry. So instead of comparing 100,000 geometries vs another 100,000. I union the first geometry to a single geometry and compare 1 vs 100,000. I create a PreparedGeometry once on the single unioned geometry.
